packages/dev/core/src/Animations/animation.optimizations.ts

```diff
@@ -0,0 +1,67 @@
+import type { Scene } from "core/scene";
+
+/**
+ * Interface used to define the optimization options for animations
+ */
+export interface IAnimationOptimization {
+    /**
+     * If true, all runtime animations will be merged into the first animatable
+     * @defaultValue true
+     */
+    mergeRuntimeAnimations: boolean;
+    /**
+     * If true, all keyframes evaluation will be merged from the first runtime animation
+     * You need to turn on `mergeRuntimeAnimations` for this to work
+     * @defaultValue false
+     */
+    mergeKeyFrames: boolean;
+}
+
+/**
+ * This is a destructive optimization that merges all animatables into the first one.
+ * That animatable will also host all the runtime animations.
+ * We expect that all the animatables are on the same timeframe (same start, end, loop, etc..)
+ * @param scene defines the scene to optimize
+ * @param options defines the optimization options
+ */
+export function OptimizeAnimations(scene: Scene, options: Partial<IAnimationOptimization> = {}) {
+    const mergeRuntimeAnimations = options.mergeRuntimeAnimations ?? true;
+    const mergeKeyFrames = options.mergeKeyFrames ?? false;
+
+    // We will go through all the current animatables and merge them
+    const animatables = scene.animatables;
+
+    if (animatables.length === 0) {

```

Comment:
`animatables.length === 0` this could be replaced with `animatables.length <= 1`?

Reply:
And please check my PR :) others approved it but they are waiting your input https://github.com/BabylonJS/Babylon.js/pull/16484

Reply:
no we can still have one single animatable but plenty of runtimeAnimations in it